### ISSUE TITLE
fragment-to-light varyings segfault on transparent shaders (NVIDIA 456.71)

### ISSUE THREAD

**Issue by lyuma** [godotengine/godot#48034]
**Godot version:**
4.0.dev f817e7fb

**OS/device including version:**
Windows 10.0.19041.867
Nvidia RTX 2080 Ti
Driver version 456.71

**Issue description:**
If I use `varying` syntax to transfer data from fragment() to light(), and I also write to ALPHA, as in the following shader:
```glsl
shader_type spatial;

varying vec2 some_varying;

void fragment() {
        some_varying = vec2(0.0);
        ALPHA = 1.0;
}

void light() {
        vec2 unused = some_varying;
}
```
If this shader is used, I get the following crash:
```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] <couldn't map PC to fn name>
[1] <couldn't map PC to fn name>
[2] <couldn't map PC to fn name>
[3] <couldn't map PC to fn name>
[4] <couldn't map PC to fn name>
[5] <couldn't map PC to fn name>
[6] vk_optimusGetInstanceProcAddr
[7] vkGetInstanceProcAddr
[8] vk_optimusGetInstanceProcAddr
[9] vk_optimusGetInstanceProcAddr
[10] vkGetInstanceProcAddr
[11] vkGetInstanceProcAddr
[12] vkGetInstanceProcAddr
[13] vkGetInstanceProcAddr
[14] RenderingDeviceVulkan::render_pipeline_create (S:\repo\godot-pr-pr\drivers\vulkan\rendering_device_vulkan.cpp:5697)
[15] PipelineCacheRD::_generate_version (S:\repo\godot-pr-pr\servers\rendering\renderer_rd\pipeline_cache_rd.cpp:42)
[16] RendererSceneRenderImplementation::RenderForwardClustered::_render_list_template<0> (S:\repo\godot-pr-pr\servers\rendering\renderer_rd\forward_clustered\render_forward_clustered.cpp:443)
[17] RendererSceneRenderImplementation::RenderForwardClustered::_render_list_with_threads (S:\repo\godot-pr-pr\servers\rendering\renderer_rd\forward_clustered\render_forward_clustered.cpp:532)
[18] RendererSceneRenderImplementation::RenderForwardClustered::_render_scene (S:\repo\godot-pr-pr\servers\rendering\renderer_rd\forward_clustered\render_forward_clustered.cpp:1480)
[19] RendererSceneRenderRD::render_scene (S:\repo\godot-pr-pr\servers\rendering\renderer_rd\renderer_scene_render_rd.cpp:3621)
[20] RendererSceneCull::_render_scene (S:\repo\godot-pr-pr\servers\rendering\renderer_scene_cull.cpp:2787)
[21] RendererSceneCull::render_camera (S:\repo\godot-pr-pr\servers\rendering\renderer_scene_cull.cpp:2167)
[22] RendererViewport::_draw_3d (S:\repo\godot-pr-pr\servers\rendering\renderer_viewport.cpp:88)
[23] RendererViewport::_draw_viewport (S:\repo\godot-pr-pr\servers\rendering\renderer_viewport.cpp:136)
[24] RendererViewport::draw_viewports (S:\repo\godot-pr-pr\servers\rendering\renderer_viewport.cpp:567)
[25] RenderingServerDefault::_draw (S:\repo\godot-pr-pr\servers\rendering\rendering_server_default.cpp:115)
[26] RenderingServerDefault::draw (S:\repo\godot-pr-pr\servers\rendering\rendering_server_default.cpp:387)
[27] Main::iteration (S:\repo\godot-pr-pr\main\main.cpp:2516)
[28] OS_Windows::run (S:\repo\godot-pr-pr\platform\windows\os_windows.cpp:622)
[29] widechar_main (S:\repo\godot-pr-pr\platform\windows\godot_windows.cpp:163)
[30] _main (S:\repo\godot-pr-pr\platform\windows\godot_windows.cpp:187)
[31] main (S:\repo\godot-pr-pr\platform\windows\godot_windows.cpp:199)
[32] __scrt_common_main_seh (D:\agent\_work\9\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[33] BaseThreadInitThunk
-- END OF BACKTRACE --
```
The same crash does not happen if ALPHA is not used (opaque shader)

**Steps to reproduce:**
1. Open the attached project
2. Open CrashesWhenOpened.tscn (or assign a new material with varying_crash.shader)
3. *kaboom*

**Minimal reproduction project:**
[FragToLightVaryingCrash.zip](https://github.com/godotengine/godot/files/6340318/FragToLightVaryingCrash.zip)

**Comment by lyuma** [comment]
Updated NVIDIA drivers to 466.11 and this crash no longer happens.

I confirmed that alpha transparency causes a crash on 456.71 before updating the drivers (both windows 10.0.19041.867 and windows 10.0.19041.928).
And I confirmed that alpha works fine after the update to 466.11 on Windows 10.0.19041.928.

I don't know on Godot's policy on working around bugs with older drivers. My previous driver wasn't that old and will likely exist in the wild for months at least.
At the very least, it is helpful to document this issue with transparent shaders exists for older drivers, given that it could be reproduced with only a few lines of shader code.
